File: Park/CodeTree/K개 중 하나를 N번 선택하기(Simple)/k개 중에 1개를 n번 뽑기.py
# 입력을 받아 변수에 저장
k, n = tuple(map(int, input().split()))
selected_nums = []

# 선택된 숫자들을 출력
def print_permutation():
    for num in selected_nums:
        print(num, end=" ")
    print()

# 순열을 찾는 함수
def find_permutations(cnt):
    if cnt == n:
        print_permutation()
        return
    
    for i in range(1, k + 1):
        selected_nums.append(i)
        find_permutations(cnt + 1)
        selected_nums.pop()

# 순열 찾기 시작
find_permutations(0)

File: jhyun/Backtracking/K개 중 하나를 N번 선택하기(Simple)/k개_중에_1개를_n번_뽑기.java
import java.util.*;
import java.io.*;

public class Main {
    static int k, n;
    static List<Integer> selectedNums = new ArrayList<>();

    static void printPermutation() {
        for (int i = 0; i < selectedNums.size(); i++) {
            System.out.print(selectedNums.get(i) + " ");
        }
        System.out.println();
    }

    static void findPermutations(int cnt) {
        if (cnt == n) {
            printPermutation();
            return;
        }

        for (int i = 1; i <= k; i++) {
            selectedNums.add(i);
            findPermutations(cnt + 1);
            selectedNums.remove(selectedNums.size() - 1);
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        k = Integer.parseInt(st.nextToken());
        n = Integer.parseInt(st.nextToken());

        findPermutations(0);
    }
}